frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -1526,6 +1527,65 @@
             }
         };
 
+        //  Edit Challenge Date
+        vm.challengeDateDialog = function(ev) {
+        	vm.tempChallenge_start = vm.page.start_date;
+        	vm.tempChallenge_end = vm.page.end_date;
+            $mdDialog.show({
+                scope: $scope,
+                preserveScope: true,
+                targetEvent: ev,
+                templateUrl: 'dist/views/web/challenge/edit-challenge/edit-challenge-date.html',
+                escapeToClose: false
+            });
+        };
+
+        vm.editChallengeDate = function(editChallengeDateForm) {
+            if (editChallengeDateForm) {
+                var challengeHostList = utilities.getData("challengeCreator");
+                for (var challenge in challengeHostList) {
+                    if (challenge == vm.challengeId) {
+                        vm.challengeHostId = challengeHostList[challenge];
+                        break;
+                    }
+                }
+                parameters.url = "challenges/challenge_host_team/" + vm.challengeHostId + "/challenge/" + vm.challengeId;
+                parameters.method = 'PATCH';
+                const start = new Date(vm.page.start_date).valueOf()
+                const end = new Date(vm.page.end_date).valueOf()
+                if (start < end){
+
+                parameters.data = {
+                    "start_date": vm.page.start_date,
+                    "end_date": vm.page.end_date
+                };
+                parameters.callback = {
+                    onSuccess: function(response) {
+                        var status = response.status;
+                        if (status === 200) {
+                            $mdDialog.hide();
+                            $rootScope.notify("success", "The challenge Date is  successfully updated!");
```

Comment:
Please add different messages for editing challenge start and end date. Something like
`The challenge start date is updated.` when start date is updated and same for end date.